File: backend/snippets/interactionLLM/llm.py
import requests
from gigachat import GigaChat
import requests
import uuid
import json
import os
import re
from IPython.display import display, Markdown
from dotenv import load_dotenv
from gigachat import GigaChat
class LLM:
    """
      Выполняет POST-запрос к эндпоинту, который выдает токен.
      Параметры:
      - auth_token (str): токен авторизации, необходимый для запроса.
      - область (str): область действия запроса API. По умолчанию — «GIGACHAT_API_PERS».
      Возвращает:
      - ответ API, где токен и срок его "годности".
    """
    @staticmethod
    def get_gigachat_token(auth_key, scope='GIGACHAT_API_PERS', model='GigaChat:latest'):
        giga = GigaChat(
            credentials=auth_key,
            scope=scope,
            model=model,
            #FIXME !!!!!!!!!!!!!!!!!!!!!!!!!!!!!установить сертификат
            verify_ssl_certs=False,
        )
        try:
            response = giga.get_token()
            return response.access_token
        except requests.RequestException as e:
            return None



    """
    Отправляет POST-запрос к API чата для получения ответа от модели GigaChat.

    Параметры:
    - auth_token (str): Токен для авторизации в API.
    - user_message (str): Сообщение от пользователя, для которого нужно получить ответ.

    Возвращает:
    - str: Ответ от API в виде текстовой строки.
    """
    @staticmethod
    def talk_to_chat(auth_token, user_message, model='GigaChat:latest'):
        giga = GigaChat(access_token=auth_token,
                        model=model,
                        verify_ssl_certs=False
                        )
        try:
            response = giga.chat(user_message)
            return response.choices[0].message.content
        except requests.RequestException as e:
            return None

File: src/main/java/com/aitherapist/aitherapist/interactionWithGigaApi/llm/Llm.java
package com.aitherapist.aitherapist.interactionWithGigaApi.llm;

import chat.giga.client.auth.AuthClient;
import chat.giga.client.auth.AuthClientBuilder;
import chat.giga.http.client.HttpClientException;
import chat.giga.model.*;
import chat.giga.client.*;
import chat.giga.model.completion.ChatMessage;
import chat.giga.model.completion.CompletionRequest;
import chat.giga.model.completion.CompletionResponse;

import java.util.List;


public final class Llm {
    public static String talkToChat(String accessToken, List<ChatMessage> messages) {
        GigaChatClient client = GigaChatClient.builder()
                .authClient(AuthClient.builder()
                        .withProvidedTokenAuth(accessToken).build())
                .build();
        try {
            CompletionResponse response = client.completions(CompletionRequest.builder()
                    .model(ModelName.GIGA_CHAT)
                    .messages(messages)
                    .build());
            return response.choices().get(0).message().content();
        } catch (HttpClientException ex) {
            return ex.statusCode() + "  " + ex.bodyAsString();
        }
    }


    public static String talkToChat(String accessToken, List<ChatMessage> messages, int model) {
        String modelName;
        switch (model) {
            case 1:
                modelName = ModelName.GIGA_CHAT;
                break;
            case 2:
                modelName = ModelName.GIGA_CHAT_PRO;
                break;
            case 3:
                modelName = ModelName.GIGA_CHAT_MAX;
                break;
            default:
                return "Ошибка: неизвестный номер модели. Допустимые значения: 1, 2 или 3.";
        }
        GigaChatClient client = GigaChatClient.builder()
                .authClient(AuthClient.builder()
                        .withProvidedTokenAuth(accessToken).build())
                .build();
        try {
            CompletionResponse response = client.completions(CompletionRequest.builder()
                    .model(modelName)
                    .messages(messages)
                    .build());
            return response.choices().get(0).message().content();
        } catch (HttpClientException ex) {
            return ex.statusCode() + " " + ex.bodyAsString();
        }
    }

    public static String getGigaChatToken() {
        try {
            String authKey = System.getenv("GIGA_CHAT_API_KEY");
            AuthClient client = AuthClient.builder().withOAuth(AuthClientBuilder.OAuthBuilder.builder()
                    .scope(Scope.GIGACHAT_API_PERS)
                    .authKey(authKey)
                    .build()).build();
            return client.getToken().token();
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }

    public static String getGigaChatToken(Scope scope) {
        try {
            String authKey = System.getenv("GIGA_CHAT_API_KEY");
            AuthClient client = AuthClient.builder().withOAuth(AuthClientBuilder.OAuthBuilder.builder()
                                    .scope(scope)
                                    .authKey(authKey)
                                    .build()).build();
            return client.getToken().token();
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }

}





